Sources/NIOHTTP2/HTTP2FrameParser.swift

```diff
@@ -0,0 +1,835 @@
+                var slice = bytes.readSlice(length: required)!
+                currentBytes.write(buffer: &slice)
+                return try self.readFrame(withHeader: header, from: &currentBytes)
+            } else {
+                currentBytes.write(buffer: &bytes)  // consume entire input buffer
+                return nil  // no frame (yet)
+            }
+        case .simulatingDataFrames(let header, let padding, let remaining):
+            let streamID = HTTP2StreamID(knownID: Int32(header.rawStreamID & ~0x8000_0000))
+            let flags = HTTP2Frame.FrameFlags(rawValue: header.flags)
+            if bytes.readableBytes < remaining {
+                // ping out a fake DATA frame containing these bytes, without the END_STREAM flag being set
+                let slice = bytes
+                bytes.clear()   // consuming all the bytes
+                self.state = .simulatingDataFrames(originalHeader: header, padding: padding, remainingBytes: remaining - slice.readableBytes)
+                return HTTP2Frame(streamID: streamID, flags: [], payload: .data(.byteBuffer(slice)))
+            } else {
+                // we have all remaining bytes, so we include .endStream bit (if set in original frame)
+                var slice = bytes.readSlice(length: remaining)!
+                if padding > 0 {
+                    // trim padding from the bytes we attach to the parsed frame
+                    slice.moveWriterIndex(to: slice.writerIndex - Int(padding))
+                }
+                self.state = .idle
+                return HTTP2Frame(streamID: streamID, flags: flags, payload: .data(.byteBuffer(slice)))
+            }
+        case .accumulatingHeaderBlockFragments(let header, var buffer):
+            // special case since we only accept CONTINUATION frames in this state
+            guard let continuationHeader = bytes.readFrameHeader() else {
+                return nil      // don't have another header yet
+            }
+            
+            guard continuationHeader.type == 0x9 /* CONTINUATION */ else {
+                // invalid frame: this is a connection error
+                throw NIOHTTP2Errors.ConnectionError(code: .protocolError)
+            }
+            
+            let flags = HTTP2Frame.FrameFlags(rawValue: continuationHeader.flags)
+            
+            var cbytes: ByteBuffer
+            if bytes.readableBytes >= continuationHeader.length {
+                cbytes = bytes.readSlice(length: continuationHeader.length)!
+                buffer.write(buffer: &cbytes)
+                
+                if (flags.contains(.endHeaders)) {
+                    // we're done: send out a frame
+                    return try self.emitSyntheticHeadersFrame(header, bytes: &buffer)
+                } else {
+                    self.state = .accumulatingHeaderBlockFragments(header, buffer)
+                }
+            } else {
+                cbytes = bytes
+                bytes.clear()
```

Comment:
Don't call `clear` here, it'll trigger a CoW. Use `readSlice` to consume the readable bytes instead.